src/plugins/interactive_setup/server/elasticsearch_service.test.ts

```diff
@@ -64,6 +64,18 @@ describe('ElasticsearchService', () => {
           headers: { 'x-elastic-product': 'Elasticsearch' },
         })
       );
+      mockAuthenticateClient.asInternalUser.nodes.info.mockResolvedValue(
+        interactiveSetupMock.createApiResponse({
+          statusCode: 200,
+          body: {
+            nodes: [
+              {
+                version: '8.0.0',
+              },
+            ],
+          } as any,
+        })
+      );
```

Comment:
A couple of nits:
1. We want to avoid using `any` whenever possible.
2. This technically works, but the mocks here rely on implementation details of the `pollEsNodesVersion` API. It would be better to mock that entire function instead, IMO. We have a pattern to do this with Jest, for example:

Create `elasticsearch_service.test.mocks.ts`

```ts
import type { pollEsNodesVersion } from '../../../../src/core/server';

export const mockPollEsNodesVersion = jest.fn() as jest.MockedFunction<typeof pollEsNodesVersion>;

jest.mock('../../../../src/core/server', () => ({
  pollEsNodesVersion: mockPollEsNodesVersion,
}));
```

Then you can change your tests accordingly:

```diff
diff --git a/src/plugins/interactive_setup/server/elasticsearch_service.test.ts b/src/plugins/interactive_setup/server/elasticsearch_service.test.ts
index e616517950d..ef564fe2d63 100644
--- a/src/plugins/interactive_setup/server/elasticsearch_service.test.ts
+++ b/src/plugins/interactive_setup/server/elasticsearch_service.test.ts
@@ -6,12 +6,17 @@
  * Side Public License, v 1.
  */
 
+// eslint-disable-next-line import/order
+import { mockPollEsNodesVersion } from './elasticsearch_service.test.mock';
+
 import { errors } from '@elastic/elasticsearch';
+import { BehaviorSubject } from 'rxjs';
 import tls from 'tls';
 
 import { nextTick } from '@kbn/test/jest';
 import { elasticsearchServiceMock, loggingSystemMock } from 'src/core/server/mocks';
 
+import type { NodesVersionCompatibility } from '../../../../src/core/server';
 import { ElasticsearchConnectionStatus } from '../common';
 import { ConfigSchema } from './config';
 import type { ElasticsearchServiceSetup } from './elasticsearch_service';
@@ -30,6 +35,11 @@ describe('ElasticsearchService', () => {
     mockElasticsearchPreboot = elasticsearchServiceMock.createPreboot();
   });
 
+  function mockCompatibility(isCompatible: boolean, message?: string) {
+    mockPollEsNodesVersion.mockReturnValue(
+      new BehaviorSubject({ isCompatible, message } as NodesVersionCompatibility).asObservable()
+    );
+  }
   describe('#setup()', () => {
     let mockConnectionStatusClient: ReturnType<
       typeof elasticsearchServiceMock.createCustomClusterClient
@@ -64,18 +74,7 @@ describe('ElasticsearchService', () => {
           headers: { 'x-elastic-product': 'Elasticsearch' },
         })
       );
-      mockAuthenticateClient.asInternalUser.nodes.info.mockResolvedValue(
-        interactiveSetupMock.createApiResponse({
-          statusCode: 200,
-          body: {
-            nodes: [
-              {
-                version: '8.0.0',
-              },
-            ],
-          } as any,
-        })
-      );
+      mockCompatibility(true);
 
       setupContract = service.setup({
         elasticsearch: mockElasticsearchPreboot,
@@ -412,18 +411,7 @@ describe('ElasticsearchService', () => {
           interactiveSetupMock.createApiResponse({ statusCode: 200, body: {} as any })
         );
 
-        mockAuthenticateClient.asInternalUser.nodes.info.mockResolvedValue(
-          interactiveSetupMock.createApiResponse({
-            statusCode: 200,
-            body: {
-              nodes: [
-                {
-                  version: '7.0.0',
-                },
-              ],
-            } as any,
-          })
-        );
+        mockCompatibility(false, 'Oh no!');
 
         await expect(
           setupContract.enroll({
@@ -431,9 +419,7 @@ describe('ElasticsearchService', () => {
             hosts: ['host1', 'host2'],
             caFingerprint: 'DE:AD:BE:EF',
           })
-        ).rejects.toMatchInlineSnapshot(
-          `[CompatibilityError: This version of Kibana (v8.0.0) is incompatible with the following Elasticsearch nodes in your cluster: v7.0.0 @ undefined (undefined)]`
-        );
+        ).rejects.toMatchInlineSnapshot(`[CompatibilityError: Oh no!]`);
 
         // Check that we properly closed all clients.
         expect(mockEnrollClient.close).toHaveBeenCalledTimes(1);
@@ -575,24 +561,11 @@ some weird+ca/with
           interactiveSetupMock.createApiResponse({ statusCode: 200, body: true })
         );
 
-        mockAuthenticateClient.asInternalUser.nodes.info.mockResolvedValue(
-          interactiveSetupMock.createApiResponse({
-            statusCode: 200,
-            body: {
-              nodes: [
-                {
-                  version: '7.0.0',
-                },
-              ],
-            } as any,
-          })
-        );
+        mockCompatibility(false, 'Oh no!');
 
         await expect(
           setupContract.authenticate({ host: 'http://localhost:9200' })
-        ).rejects.toMatchInlineSnapshot(
-          `[CompatibilityError: This version of Kibana (v8.0.0) is incompatible with the following Elasticsearch nodes in your cluster: v7.0.0 @ undefined (undefined)]`
-        );
+        ).rejects.toMatchInlineSnapshot(`[CompatibilityError: Oh no!]`);
         expect(mockAuthenticateClient.close).toHaveBeenCalledTimes(1);
       });
```

And that has the added benefit of simplifying the tests a bit :D

Reply:
Nice, thanks, will change this.